File: backend/letscookLibrary/letscookLibrary/Cook.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace letscookLibrary
{
    public class Cook: BaseEntity
    {
        public string Login { get; set; }
        public string Password { get; set; }
        public IList<Ingredient> Inventory { get; set; }
        public IList<Recipe> Recipes { get; set; }
        public IList<ShoppingListItem> ShoppingList { get; set; }
        public Setting Settings { get; set; }

        public void AddRecipe(Recipe value)
        {
            value.Id = Guid.NewGuid();
            Recipes.Add(value);
        }
    }
}

File: frontend/nativescript/src/app/model/cook.ts
import { BaseEntity } from "./base-entity";
import { Recipe } from "./recipe";
import { Settings } from "./settings";
import { ShoppingListItem } from "./shopping-list-item";
import { Ingredient } from "./ingredient";

export class Cook extends BaseEntity {
    login: string;
    password: string;
    inventory: Array<Ingredient>;
    recipes: Array<Recipe>;
    shoppingList: Array<ShoppingListItem>;
    settings: Settings;
}
